gammapy/spectrum/models.py

```diff
@@ -1063,31 +1065,174 @@ def plot(self, energy_range, ax=None, energy_unit='TeV',
         return ax
 
 
-class AbsorbedSpectralModel(SpectralModel):
+class Absorption(object):
+    """
+    Class dealing with absorption model
+
+    Parameters
+    ----------
+    energy_lo : `~astropy.units.Quantity`
+        Lower bin edges of energy axis
+    energy_hi : `~astropy.units.Quantity`
+        Upper bin edges of energy axis
+    param_lo : `~astropy.units.Quantity`
+        Lower bin edges of parameter axis
+    param_hi : `~astropy.units.Quantity`
+        Upper bin edges of parameter axis
+    data : `~astropy.units.Quantity`
+        Model value
+
+    Examples
+    --------
+    Create and plot EBL absorption model for a redshift of 0.2
+
+    .. create::
+        :include-source:
+        import matplotlib.pyplot as plt
+        from gammapy.spectrum.models import Absorption
 
-    def __init__(self, spectral_model, table_model):
-        """Absorbed spectral model
+        table = Absorption.read(filename='$GAMMAPY_EXTRA/datasets/ebl/ebl_dominguez11.fits.gz').table_model(parameter=0.2)
+
+        ax = plt.gca()
+        opts = dict(energy_range=[0.01, 100] * u.TeV, energy_unit='TeV', ax=ax)
+        table.plot(label='Dominguez 2011', **opts)
+        ax.set_ylabel(r'Absorption coefficient [$\exp{(-\tau(E))}$]')
+        ax.set_yscale('log')
+        ax.set_ylim([1.e-4, 2.])
+        plt.grid(which='both')
+        plt.legend(loc='best')
+        plt.show()
+
+    """
+
+    def __init__(self, energy_lo, energy_hi, param_lo, param_hi, data):
+        axes = [
+            BinnedDataAxis(param_lo, param_hi,
+                           interpolation_mode='linear', name='parameter'), 
+            BinnedDataAxis(energy_lo,energy_hi,
+                           interpolation_mode='log', name='energy')
+        ]
+
+        self.absorption = NDDataArray(axes=axes, data=data)
```

Comment:
Suggest to rename `self.absorption` to `self.data` here.
Otherwise you get `absorption.absorption` in code that uses this class, which is confusing.